virtualmachine/google/util.go

```
@@ -0,0 +1,416 @@
+				Boot:       true,
+				AutoDelete: svc.vm.AutoDelete,
+				InitializeParams: &gce.AttachedDiskInitializeParams{
+					SourceImage: image.SelfLink,
+					DiskSizeGb:  int64(svc.vm.DiskSize),
+					DiskType:    fmt.Sprintf("zones/%s/diskTypes/%s", zone.Name, svc.vm.DiskType),
+				},
+			},
+		},
+		MachineType: machineType.SelfLink,
+		Metadata: &gce.Metadata{
+			Items: []*gce.MetadataItems{
+				{
+					Key:   "sshKeys",
+					Value: &md,
+				},
+			},
+		},
+		NetworkInterfaces: []*gce.NetworkInterface{
+			&gce.NetworkInterface{
+				AccessConfigs: []*gce.AccessConfig{
+					&accessconfig,
+				},
+				Network:    network.SelfLink,
+				Subnetwork: subnetworkSelfLink,
+			},
+		},
+		Scheduling: &gce.Scheduling{
+			Preemptible: svc.vm.Preemptible,
+		},
+		ServiceAccounts: []*gce.ServiceAccount{
+			&gce.ServiceAccount{
+				Email:  "default",
+				Scopes: svc.vm.Scopes,
+			},
+		},
+		Tags: &gce.Tags{
+			Items: svc.vm.Tags,
+		},
+	}
+
+	op, err := svc.service.Instances.Insert(svc.vm.Project, zone.Name, instance).Do()
+	if err != nil {
+		return err
+	}
+
+	if err = svc.waitForOperationReady(op.Name); err != nil {
+		return err
+	}
+
+	_, err = svc.getInstance()
+	return err
+}
+
+// starts an stopped GCE instance.
+func (svc *googleService) start() error {
+	instance, err := svc.getInstance()
+	if err != nil {
+		if !strings.Contains(err.Error(), "no instance found") {
+			return err
+		}
+	}
+
+	if instance == nil {
+		return errors.New("no instance found")
+	}
+
+	op, err := svc.service.Instances.Start(svc.vm.Project, svc.vm.Zone, svc.vm.Name).Do()
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("Waiting for instance to start")
+	return svc.waitForOperationReady(op.Name)
+}
+
+// starts an stopped GCE instance.
+func (svc *googleService) stop() error {
+	instance, err := svc.getInstance()
+	if err != nil {
+		if !strings.Contains(err.Error(), "no instance found") {
+			return err
+		}
+	}
+
+	if instance == nil {
```

Can you explain this logic? You're ignoring errors where the error contains "no instance found" but if instance is nil, you return "no instance found"? A comment would be useful